### PR TITLE
RF: dcm2niix interface

### PR DESCRIPTION
**Pull request by mgxd**
## Summary

This PR fixes the convoluted approach used for gathering `dcm2niix` outputs.

## List of changes proposed in this PR (pull-request)

- Checks the output prefix for possible suffixes, instead of assuming outputs based off stdout switches.

## Acknowledgment

- [x] \(Mandatory\) I acknowledge that this contribution will be available under the Apache 2 license.

### CODE REVIEW

File: nipype/interfaces/dcm2nii.py [hunk 1]
```diff
@@ -410,52 +412,44 @@ def _run_interface(self, runtime):
...
-        else:
-            return files, bvecs, bvals, bids
+            if line.startswith("Convert "):  # output
+                fname = str(re.search('\S+/\S+', line).group(0))
+                filenames.append(os.path.abspath(fname))
+        return filenames
+
+    def _parse_files(self, filenames):
+        outfiles, bvals, bvecs, bids = [], [], [], []
+        outtypes = (".nii", ".nii.gz", ".bval", ".bvec", ".json")
```
Comment (mgxd) [inline nipype/interfaces/dcm2nii.py]: @neurolabusc not that we were catching it before, but am I missing any output formats?

Comment (neurolabusc) [reply nipype/interfaces/dcm2nii.py]: The recent release (v1.0.20190410) includes ".nrrd" (`-e y -z n`) and ". nhdr"/".raw.gz" (`-e y -z y`). This was added for the Slicer team and so you may or may not want to support NRRD format export. There is also a deprecated text output  ".txt" (`-t y`) which predates BIDS.

**Comment by effigies** [comment]
@mgxd Any chance of a test? This is pretty tough to evaluate without seeing the code paths exercised.

**Comment by mgxd** [comment]
we already have a regression test
https://github.com/nipy/nipype/blob/704b97dee7848283692bac38f04541c5af2a87b5/nipype/interfaces/tests/test_extra_dcm2nii.py#L27
though I'm not sure it's actually being hit in the current testing. I'll fix this up for the next release

**Comment by oesteban** [comment]
CircleCI (both py36 and py27):
```
[gw1] [ 87%] SKIPPED ../src/nipype/interfaces/tests/test_extra_dcm2nii.py::test_dcm2niix_dwi 
```

Travis (py37):
```
interfaces/tests/test_extra_dcm2nii.py::test_dcm2niix_dwi 
[gw1] [ 87%] SKIPPED interfaces/tests/test_extra_dcm2nii.py::test_dcm2niix_dwi 
```

**Comment by mgxd** [comment]
looks like we never got around to adding it to testing in #2498 - looks like the original problem (https://github.com/nipy/nipype/pull/2498#issuecomment-373871577) was fixed so I'll add that change here.

**Comment by effigies** [comment]
@mgxd I think you need to merge/rebase master to fix the Docker builds.

**Comment by effigies** [comment]
@mgxd What's the status of this one?

**Comment by mgxd** [comment]
ready for review!

**Pull request merged**